extensions/sql-database-projects/src/controllers/databaseProjectTreeViewProvider.ts

```diff
@@ -58,4 +59,20 @@ export class SqlDatabaseProjectTreeViewProvider implements vscode.TreeDataProvid
 		this.roots = newRoots;
 		this._onDidChangeTreeData.fire(undefined);
 	}
+
+	public setTreeView(value: vscode.TreeView<BaseProjectTreeItem>) {
+		if (this.treeView) {
+			throw new Error('TreeView should not be set multiple times.');
```

Comment:
it is feasible to move this to the constructor?

Reply:
Don't think so; there's a chicken-and-egg problem (check mainController.ts:68).

Basically, the treeView, understandably, requires a treeViewProvider as part of its constuctor, but I also want my treeViewProvider to be able to direct the treeView to focus on something in it.

If it were possible to get the treeViewProvider's parent treeView, then this wouldn't be an issue, so I think that's a bit of an oversight on vscode's side.